<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Zsombor Gegesy</author>
 <class>RuleEditor</class>
 <widget class="QDialog" name="RuleEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>460</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Network Access Rules</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="adblockerGroupBox">
     <property name="title">
      <string>AdBlocker Enabled</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QTableView" name="ruleListView">
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>50</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="1" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Rule:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="ruleEdit"/>
        </item>
        <item row="2" column="1">
         <widget class="QCheckBox" name="regExpCheckBox">
          <property name="text">
           <string>Regular expression</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QCheckBox" name="exclusionCheckBox">
          <property name="toolTip">
           <string>If checked, then this will be an exception rule, network access granted</string>
          </property>
          <property name="text">
           <string>Access Granted</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="addButton">
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="modifyButton">
            <property name="text">
             <string>Modify</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="removeButton">
            <property name="text">
             <string>Remove</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="4" column="1">
         <widget class="QCheckBox" name="enabledCheckBox">
          <property name="text">
           <string>Enabled</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>ruleEdit</tabstop>
  <tabstop>regExpCheckBox</tabstop>
  <tabstop>exclusionCheckBox</tabstop>
  <tabstop>addButton</tabstop>
  <tabstop>modifyButton</tabstop>
  <tabstop>removeButton</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RuleEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>450</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RuleEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>450</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RuleEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>450</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>-2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleEditor</receiver>
   <slot>addRule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>129</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modifyButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleEditor</receiver>
   <slot>modifyRule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>340</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleEditor</receiver>
   <slot>deleteRule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>488</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addRule()</slot>
  <slot>modifyRule()</slot>
  <slot>deleteRule()</slot>
 </slots>
</ui>
